### ISSUE TITLE
Error while fetching your routes ! (See output console for more details)

### ISSUE THREAD

**Issue by macro0000** [Julien-R44/adonis-vscode-extension#39]
![image](https://github.com/Julien-R44/adonis-vscode-extension/assets/25545246/503096e0-4c83-424a-b406-168fff119f4e)

```
Error: Command failed with exit code 1: node --dns-result-order=ipv4first ace list:routes --json
node:events:497
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at target._send (node:internal/child_process:879:20)
    at target.send (node:internal/child_process:752:19)
    at Application.notify (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+application@8.2.2_@adonisjs+config@5.0.2_@adonisjs+fold@10.1.2/node_modules/@adonisjs/application/build/chunk-WIOBO4NG.js:1171:15)
    at Application.start (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+application@8.2.2_@adonisjs+config@5.0.2_@adonisjs+fold@10.1.2/node_modules/@adonisjs/application/build/chunk-WIOBO4NG.js:1317:10)
    at async file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+core@6.9.0_@adonisjs+assembler@7.6.1_@vinejs+vine@2.0.0/node_modules/@adonisjs/core/build/src/ignitor/ace.js:59:17
    at async #exec (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@poppinss+hooks@7.2.3/node_modules/@poppinss/hooks/build/index.js:77:24)
    at async Kernel.find (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+ace@13.0.0/node_modules/@adonisjs/ace/build/index.js:1728:5)
    at async #execMain (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+ace@13.0.0/node_modules/@adonisjs/ace/build/index.js:1450:23)
    at async AceProcess.handle (file:///D:/project-myself/server_wx_store/node_modules/.pnpm/@adonisjs+core@6.9.0_@adonisjs+assembler@7.6.1_@vinejs+vine@2.0.0/node_modules/@adonisjs/core/build/src/ignitor/ace.js:66:9)
Emitted 'error' event on process instance at:
    at node:internal/child_process:883:39
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11) {
  errno: -4047,
  code: 'EPIPE',
  syscall: 'write'
}

Node.js v21.7.2

```

**Comment by mattsimonis** [comment]
Also running into this. Would love a fix.

**Comment by mattsimonis** [comment]
My issue ended up being a logger.info that weirdly only appears when the list:routes is run through this extension. So it threw off the JSON parsing because of a log at the tail.

**Comment by ajayhada** [comment]
Error: Command failed with exit code 1: node --dns-result-order=ipv4first ace list:routes --json

node:internal/modules/run_main:129
    triggerUncaughtException(
    ^
Error: Failed to load native binding
    at Object.<anonymous> (c:\Projects\hello-world\node_modules\@swc\core\binding.js:333:11)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Object..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Function._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (c:\Projects\hello-world\node_modules\@swc\core\index.js:49:17)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Object..js (node:internal/modules/cjs/loader:1416:10) {
  [cause]: [
    Error: Cannot find module './swc.win32-x64-msvc.node'
    Require stack:
    - c:\Projects\hello-world\node_modules\@swc\core\binding.js
    - c:\Projects\hello-world\node_modules\@swc\core\index.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\transpilers\swc.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\index.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\esm.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\esm.mjs
        at Function.<anonymous> (node:internal/modules/cjs/loader:1145:15)
        at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (c:\Projects\hello-world\node_modules\@cspotcode\source-map-support\source-map-support.js:811:30)
        at Function._load (node:internal/modules/cjs/loader:986:27)
        at Module.require (node:internal/modules/cjs/loader:1233:19)
        at require (node:internal/modules/helpers:179:18)
        at requireNative (c:\Projects\hello-world\node_modules\@swc\core\binding.js:91:16)
        at Object.<anonymous> (c:\Projects\hello-world\node_modules\@swc\core\binding.js:306:17)
        at Module._compile (node:internal/modules/cjs/loader:1358:14)
        at Object..js (node:internal/modules/cjs/loader:1416:10)
        at Module.load (node:internal/modules/cjs/loader:1208:32),
    Error: Cannot find module '@swc/core-win32-x64-msvc'
    Require stack:
    - c:\Projects\hello-world\node_modules\@swc\core\binding.js
    - c:\Projects\hello-world\node_modules\@swc\core\index.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\transpilers\swc.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\index.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\dist\esm.js
    - c:\Projects\hello-world\node_modules\ts-node-maintained\esm.mjs
        at Function.<anonymous> (node:internal/modules/cjs/loader:1145:15)
        at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (c:\Projects\hello-world\node_modules\@cspotcode\source-map-support\source-map-support.js:811:30)
        at Function._load (node:internal/modules/cjs/loader:986:27)
        at Module.require (node:internal/modules/cjs/loader:1233:19)
        at require (node:internal/modules/helpers:179:18)
        at requireNative (c:\Projects\hello-world\node_modules\@swc\core\binding.js:96:16)
        at Object.<anonymous> (c:\Projects\hello-world\node_modules\@swc\core\binding.js:306:17)
        at Module._compile (node:internal/modules/cjs/loader:1358:14)
        at Object..js (node:internal/modules/cjs/loader:1416:10)
        at Module.load (node:internal/modules/cjs/loader:1208:32)
  ]
}

Node.js v20.16.0

**Comment by rbartholomay** [comment]
I use the logger inside a preload file. And if this preload file will do some log output the extension can´t fetch the routes.
The command `node ace list:routes` runs fine in the terminal!

Perhaps it´s an idea to disable the logger when the extension try to fetch the routes???

Bye, René

**Comment by Zenkylo** [comment]
See this in a fresh Adonis v6 app.
The routes list, but throw an error.
```bash
 hello-world  node ace list:routes --json
[
  {
    "domain": "root",
    "routes": [
      {
        "name": "",
        "pattern": "/",
        "methods": [
          "GET"
        ],
        "handler": {
          "type": "closure",
          "name": "closure"
        },
        "middleware": []
      },
      {
        "name": "",
        "pattern": "/send-email",
        "methods": [
          "GET"
        ],
        "handler": {
          "type": "controller",
          "moduleNameOrPath": "#controllers/email_controller",
          "method": "sendEmail"
        },
        "middleware": []
      }
    ]
  }
]

   Error: Cannot find package '@aws-sdk/client-ses' imported from C:\Users\kylea\Git\adonis-samples\ses-test\hello-world\node_modules\@adonisjs\mail\build\src\transports\ses.js


   ⁃ packageResolve
     node_modules\ts-node-maintained\dist-raw\node-internal-modules-esm-resolve.js:969
   ⁃ moduleResolve
     node_modules\ts-node-maintained\dist-raw\node-internal-modules-esm-resolve.js:1010
   ⁃ Object.defaultResolve
     node_modules\ts-node-maintained\dist-raw\node-internal-modules-esm-resolve.js:1125
```

Here's a simple controller, taken from the Adonis Mail part of the docs.
```typescript
// email_controller.ts
import type { HttpContext } from '@adonisjs/core/http'
import mail from '@adonisjs/mail/services/main'

export default class QueuesController {
  public async endpoint({ request, response }: HttpContext) {

    await mail.send((message) => {
      message
        .to('test@email.com')
        .from('info@example.org')
        .subject('Verify your email address')
        .htmlView('emails/verify_email')
    })

    response.status(200).send('Job added to queue')
  }
}

```

**Comment by mtariqsajid** [comment]
is there a way to turn it off, its always try to fetch the routes list json. there should a option to fetch the routes or not.

**Comment by mtariqsajid** [comment]
Hello @Julien-R44 . Please tell me if there is a way to turn off the option of the VS Code AdonisJS extension that's attempting to fetch the routes list. It keeps showing an error on every startup of a project. Additionally, I'd like to know if this extension supports the latest AdonisJS Framework version, 6.

**Comment by pumpkinlink** [comment]
I managed to work around this by manually setting the extension "Node Path" vscode setting to `/home/(myuser)/.nvm/versions/node/v22.12.0/bin/node`

**Comment by Exefer** [comment]
I'm running into the same error.

**Comment by forgiv** [comment]
tl;dr - Switching `Node Path` to a node executable that's in a user managed directory instead of a system managed directory fixed the issue for me.


So I had the same issue as OP:

```
Error: Command failed with exit code 1: node --dns-result-order=ipv4first ace list:routes --json
node:events:485
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at target._send (node:internal/child_process:877:20)
    at target.send (node:internal/child_process:751:19)
    at Application.notify (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/application/build/chunk-GCU5TAHR.js:1192:15)
    at Application.start (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/application/build/chunk-GCU5TAHR.js:1338:10)
    at async file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/core/build/src/ignitor/ace.js:59:17
    at async #exec (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@poppinss/hooks/build/index.js:77:24)
    at async Kernel.find (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/ace/build/index.js:1737:5)
    at async #execMain (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/ace/build/index.js:1452:23)
    at async AceProcess.handle (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/core/build/src/ignitor/ace.js:66:9)
Emitted 'error' event on process instance at:
    at node:internal/child_process:881:39
    at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
  errno: -4047,
  code: 'EPIPE',
  syscall: 'write'
}

Node.js v24.1.0
```

After reading the comment left by @pumpkinlink about specifying the `Node Path` I ran `(Get-Command node).Path` in PowerShell to get the path of the Node executable and used that. AND...

```
Error: Command failed with exit code 1: C:\Program Files\Volta\node.exe --dns-result-order=ipv4first ace list:routes --json
node:events:485
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at target._send (node:internal/child_process:877:20)
    at target.send (node:internal/child_process:751:19)
    at Application.notify (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/application/build/chunk-GCU5TAHR.js:1192:15)
    at Application.start (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/application/build/chunk-GCU5TAHR.js:1338:10)
    at async file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/core/build/src/ignitor/ace.js:59:17
    at async #exec (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@poppinss/hooks/build/index.js:77:24)
    at async Kernel.find (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/ace/build/index.js:1737:5)
    at async #execMain (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/ace/build/index.js:1452:23)
    at async AceProcess.handle (file:///c:/Users/shark/Dev/adonisjs-ext-test/node_modules/@adonisjs/core/build/src/ignitor/ace.js:66:9)
Emitted 'error' event on process instance at:
    at node:internal/child_process:881:39
    at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
  errno: -4047,
  code: 'EPIPE',
  syscall: 'write'
}

Node.js v24.1.0
```

Exactly the same error, but now it has the full path for the node executable.  
Then it dawned on me, I'm using Volta to manage my node versions so why is the version in my path in my `Program Files` folder instead of a somewhere in `AppData` where I would expect it to be?

So I ran `volta which node` instead and got `C:\Users\shark\AppData\Local\Volta\tools\image\node\24.1.0\node.exe`.
I then set THAT as my `Node Path` in the extension settings and... IT WORKED!

In my case, the error clearly has something to do with permissions.  
Maybe the extension is writing to a temporary buffer in the same directory as the node executable and that was causing the issue? I don't about how VS Code extensions work, so that's my best guess.

**Comment by thisisrifai** [comment]
I'm getting the same issue, please the issue appeared when started using volta (may help)

**Comment by forgiv** [comment]
> I'm getting the same issue, please note that the issue appeared when started using volta (may help)

Have you tried the solution that helped me with volta? Use `volta which node` to get the path of the node executable and set the `Node Path` setting in the extension to that path.

**Comment by mtariqsajid** [comment]
> I managed to work around this by manually setting the extension "Node Path" vscode setting to `/home/(myuser)/.nvm/versions/node/v22.12.0/bin/node`

this is working so try this solution @forgiv  @thisisrifai
